pkg/machine/qemu/virtiofsd.go

```diff
@@ -0,0 +1,116 @@
+//go:build linux || freebsd
+
+package qemu
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/podman/v5/pkg/machine/define"
+	"github.com/containers/podman/v5/pkg/machine/vmconfigs"
+	"github.com/containers/storage/pkg/fileutils"
+	"github.com/sirupsen/logrus"
+)
+
+// The preferred mechanism for configuring virtiofsd in an OS component/package
+// is to create a symlink from podman's helper directory to the desired binary.
+// This is how e.g. `catatonit` works.  However for now, we hardcode a few
+// well-known distribution paths here as a fallback.
+const virtiofsdPathFedora = "/usr/libexec/virtiofsd"
+const virtiofsdPathDebian = "/usr/lib/qemu/virtiofsd"
+const virtiofsdPathArch = "/usr/lib/virtiofsd"
+
+// VirtiofsdSpawner spawns an instance of virtiofsd
+type virtiofsdSpawner struct {
+	runtimeDir *define.VMFile
+	binaryPath string
+	mountCount uint
+}
+
+func newVirtiofsdSpawner(runtimeDir *define.VMFile) (*virtiofsdSpawner, error) {
+	var binaryPath string
+	cfg, err := config.Default()
+	if err != nil {
+		return nil, err
+	}
+	binaryPath, err = cfg.FindHelperBinary("virtiofsd", true)
+	if err != nil {
+		logrus.Debugf("Failed to find virtiofsd in helper dir, using fallback")
+		for _, p := range []string{virtiofsdPathDebian, virtiofsdPathFedora, virtiofsdPathArch} {
+			if err := fileutils.Exists(p); err == nil {
+				binaryPath = p
+				break
+			}
+		}
```

Comment:
Do we really want to do that? If we allow these paths than any distro just wants to be added here instead of configuring it once. IMO we shouldn't try these at all and require the symlink setup or require this to be set at build time and/or patched.

Reply:
> IMO we shouldn't try these at all and require the symlink setup or require this to be set at build time and/or patched.

Fair, but I think there's going to be some short term pain from that in terms of things like running podman machine from git not working unless you've got the symlink on your host's `/usr/libexec/podman` setup.  But yes, done.